src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,43 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonal = new int[matrix.length];
+        for (int indexOfMatrix = 0; indexOfMatrix < matrix.length; indexOfMatrix++) {
+            mainDiagonal[indexOfMatrix] = matrix[indexOfMatrix][indexOfMatrix];
+        }
+        return mainDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        for (int indexOfMatrix = 0; indexOfMatrix < matrix.length; indexOfMatrix++) {
+            counterDiagonal[indexOfMatrix] =
+                    matrix[indexOfMatrix][matrix.length - indexOfMatrix - 1];
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int minValue = Integer.MAX_VALUE;
```

Comment:
you can do this without Integer.MAX_VALUE - just assign first element of your matrix

Reply:
Done